type-tests/files/createSlice.typetest.ts

```diff
@@ -77,3 +77,27 @@ import {
   // typings:expect-error
   counter.actions.multiply('2')
 }
+
+
+
+/*
+ * Test: Slice action creator types properties are correct
+ */
+{
+  const counter = createSlice({
+    slice: 'counter',
+    initialState: 0,
+    reducers: {
+      increment: state => state + 1,
+      decrement: state => state - 1,
+      multiply: (state, { payload }: PayloadAction<number | number[]>) =>
+        Array.isArray(payload)
+          ? payload.reduce((acc, val) => acc * val, state)
+          : state * payload
+    }
+  })
+
+  const x: "increment" = counter.actions.increment.type;
```

Comment:
Why are these not `"counter/increment"`?  That's really confusing to me.

Reply:
Good catch.
Because I typed them wrongly and those tests only check for types at compile time, not validate that it is the real output.
That said, I'll have to remove that and only type the `type` as string then - can't do concatenation with types :(